### PR TITLE
Crosscompile for AmigaOS3 - Changes to Configure.ac and Nonblock.c

### PR DESCRIPTION
**Pull request by boingball**
Changes to Configure.ac and nonblock.c to allow the AmigaOS3 cross compile to run correctly.
Without these changes the AmigaOS3 cannot be built using configure.

However the configure.ac defines will only go in if AmiSSL is selected via this method.
so if you don't select AmiSSL I would assume the compilation would fail

With these changes, the AmigaOS 3 Cross Compile works with the following build statement

./buildconf && PKG_CONFIG=true ./configure --host=m68k-amigaos --build=$(uname -m)-pc-linux-gnu CC=m68k-amigaos-gcc AR=m68k-amigaos-ar --disable-shared --disable-ipv6 --prefix=/opt/amiga13 --disable-ntlm-wb --disable-netrc --without-libpsl --with-amissl --disable-threaded-resolver CFLAGS="-m68000 -O0 -msoft-float -Werror-implicit-function-declaration -Wno-system-headers -mcrt=clib2" CPPFLAGS="-D__AMIGA__" LIBS="-lnet -lm -lc -lz -lunix -latomic"

Tested with AmigaOS 3.2.2.1

### CODE REVIEW

**Comment by bagder** [comment]
> However the configure.ac defines will only go in if AmiSSL is selected via this method.
so if you don't select AmiSSL I would assume the compilation would fail

This seems like the wrong way, but maybe it's not worth fighting over.

File: configure.ac [hunk 1]
```diff
@@ -1338,6 +1338,9 @@ if test "$HAVE_GETHOSTBYNAME" != "1" -o "${with_amissl+set}" = set; then
     AC_MSG_RESULT([yes])
     HAVE_GETHOSTBYNAME="1"
     HAVE_PROTO_BSDSOCKET_H="1"
+    HAVE_IOCTLSOCKET_CAMEL_FIONBIO="1"
+    AC_DEFINE(HAVE_IOCTLSOCKET_CAMEL_FIONBIO, 1, [Amiga NonBlock Method])
+    AC_SUBST(HAVE_IOCTLSOCKET_CAMEL_FIONBIO, 1)
```
[the remark this answers is not in the document]
Comment (vszakats) [reply configure.ac:1343]: `IoctlSocket()` is a macro, and it fails passing the prototype check (= `cpp bsdsocket.h | grep IoctlSocket`).

I wonder if it'd be better to delete that check instead:
```diff
--- a/m4/curl-functions.m4
+++ b/m4/curl-functions.m4
@@ -2890,19 +2890,6 @@ AC_DEFUN([CURL_CHECK_FUNC_IOCTLSOCKET_CAMEL], [
   ])
   #
   if test "$tst_links_ioctlsocket_camel" = "yes"; then
-    AC_MSG_CHECKING([if IoctlSocket is prototyped])
-    AC_EGREP_CPP([IoctlSocket],[
-      $curl_includes_bsdsocket
-    ],[
-      AC_MSG_RESULT([yes])
-      tst_proto_ioctlsocket_camel="yes"
-    ],[
-      AC_MSG_RESULT([no])
-      tst_proto_ioctlsocket_camel="no"
-    ])
-  fi
-  #
-  if test "$tst_proto_ioctlsocket_camel" = "yes"; then
     AC_MSG_CHECKING([if IoctlSocket is compilable])
     AC_COMPILE_IFELSE([
       AC_LANG_PROGRAM([[
@@ -2933,7 +2920,6 @@ AC_DEFUN([CURL_CHECK_FUNC_IOCTLSOCKET_CAMEL], [
   #
   AC_MSG_CHECKING([if IoctlSocket might be used])
   if test "$tst_links_ioctlsocket_camel" = "yes" &&
-     test "$tst_proto_ioctlsocket_camel" = "yes" &&
      test "$tst_compi_ioctlsocket_camel" = "yes" &&
      test "$tst_allow_ioctlsocket_camel" = "yes"; then
     AC_MSG_RESULT([yes])
```

**Comment by jay** [comment]
I don't know if this is right or not, maybe it's just specific to this user. We don't have anyone using Amiga to confirm this is needed.

**Comment by vszakats** [comment]
I have AmigaOS builds running in CI via #15543.

It includes the `nonblock.c` patch from this PR, and 
a whole lot of other fixes.

The configuration is also simpler after reducing it
a bit: https://github.com/curl/curl/pull/15543#issuecomment-2466989712

**Comment by vszakats** [comment]
Thank you @boingball, this was merged via #15543.

**Comment by boingball** [comment]
Glad I could be of some assistance @vszakats - Thanks for sorting!

**Pull request closed without merging**